secboot/secboot_sb.go

```diff
@@ -669,3 +672,54 @@ func AddContainerRecoveryKey(devicePath string, slotName string, rkey keys.Recov
 	}
 	return sbAddLUKS2ContainerRecoveryKey(devicePath, slotName, unlockKey, sb.RecoveryKey(rkey))
 }
+
+// AddContainerRecoveryKey adds a new TPM protected key to specified device.
+//
+// Note: The unlock and primary keys are implicitly obtained from the kernel
+// keyring so this will not work if the disk was unlocked with a recovery key
+// during boot.
+func AddContainerTPMProtectedKey(devicePath, slotName string, params *ProtectKeyParams) error {
```

Comment:
I know things are a bit here and there. But maybe this function should go to `secboot_tpm.go`.